front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -249,10 +262,21 @@ export const loadAndIndexNge = async (
   // Initialize TrainrunFrequencies
   state.trainrunFrequencies = getNgeTrainrunFrequencies(state, t);
 
+  // Initialize TrainrunCategories
+  state.trainrunCategories = getTrainrunCategories(t);
+
   // Now that we have all nodes, we apply a layout
   applyLayout(state);
 };
 
+const getTrainrunCategoryId = (categoryKey?: TrainCategory | null): number => {
+  // If the categoryKey is null or undefined, we return the ID for 'NO_CATEGORY'
+  if (!categoryKey) {
+    return OSRD_TRAINRUN_CATEGORY_MAPPING.NO_CATEGORY.id;
+  }
+  return OSRD_TRAINRUN_CATEGORY_MAPPING[categoryKey].id;
+};
```

Comment:
Once OSRD_TRAINRUN_CATEGORY_MAPPING is a Map, you can simplify your function:
```suggestion
// If the categoryKey is null or undefined, we return the ID for 'NO_CATEGORY'
const getTrainrunCategoryId = (categoryKey?: TrainCategory | null) =>
  OSRD_TRAINRUN_CATEGORY_MAPPING.get(categoryKey || 'NO_CATEGORY')!.id;
```

Reply:
Done